### PR TITLE
Change error integration test to use fragment based dev-mode inference

### PR DESCRIPTION
**Pull request by cramforce**
instead of the legacy attribute based code path.

Also fixes the validator integration test to use the new validator URL and provides for a way to turn on dev mode while a page is embedded into a viewer (through the viewer).

Part of #999

### CODE REVIEW

**Comment by cramforce** [comment]
CC @jmadler

**Comment by erwinmombay** [comment]
@dvoytenko do you have any concerns?

File: src/mode.js [hunk 1]
```diff
@@ -59,11 +59,13 @@ function getMode_() {
       // occur during local dev.
       !!document.querySelector('script[src*="/dist/"],script[src*="/base/"]');
 
-  const overrideDevelopment = parseQueryString(location.hash)['development'];
+  const overrideDevelopment = parseQueryString(
+      // location.originalHash is set by the viewer when it removes the fragment
```
Comment (dvoytenko) [inline src/mode.js]: FYI, Viewer also remembers all original args in hash and can return them via `getParam(name)`. You can always use this if convenient/possible.

File: src/mode.js [hunk 2]
```diff
@@ -59,11 +59,13 @@ function getMode_() {
...
+      // from the URL.
+      location.originalHash || location.hash)['development'];
   const development = overrideDevelopment != undefined
       ? overrideDevelopment == '1'
       : !!document.querySelector('script[development]');
```
Comment (dvoytenko) [inline src/mode.js]: No longer used. Let's remove?

**Comment by dvoytenko** [comment]
Couple comments, but otherwise LGTM.

File: src/mode.js (hunk 2 above)
Comment (cramforce) [inline src/mode.js]: This CL is a step towards that.

File: src/mode.js (hunk 1 above)
Comment (cramforce) [inline src/mode.js]: Don't want mode.js depend on viewer.

**Pull request merged**